src/pages/home/report/LinkPreviewer.js

```diff
@@ -0,0 +1,142 @@
+import React, {useState, useEffect} from 'react';
+import {View, Image} from 'react-native';
+import PropTypes from 'prop-types';
+import _ from 'underscore';
+import {uniqBy} from 'lodash';
+import useWindowDimensions from '../../../hooks/useWindowDimensions';
+import Text from '../../../components/Text';
+import styles from '../../../styles/styles';
+import variables from '../../../styles/variables';
+import colors from '../../../styles/colors';
+
+const IMAGE_TYPES = ['jpg', 'jpeg', 'png'];
+const MAX_IMAGE_SIZE = 350;
+const SMALL_SCREEN_MAX_IMAGE_SIZE = 180;
+
+const propTypes = {
+    /** Data about links provided in message. */
+    linkMetadata: PropTypes.arrayOf(
+        PropTypes.shape({
+            /**  The URL of the link. */
+            url: PropTypes.string,
+
+            /**  A description of the link. */
+            description: PropTypes.string,
+
+            /**  The title of the link. */
+            title: PropTypes.string,
+
+            /**  The publisher of the link. */
+            publisher: PropTypes.string,
+
+            /**  The image associated with the link. */
+            image: PropTypes.shape({
+                /**  The height of the image. */
+                height: PropTypes.number,
+
+                /**  The width of the image. */
+                width: PropTypes.number,
+
+                /**  The URL of the image. */
+                url: PropTypes.string,
+            }),
+
+            /**  The provider logo associated with the link. */
+            logo: PropTypes.shape({
+                /**  The height of the logo. */
+                height: PropTypes.number,
+
+                /**  The width of the logo. */
+                width: PropTypes.number,
+
+                /**  The URL of the logo. */
+                url: PropTypes.string,
+            }),
+        }),
+    ),
+
+    /** Maximum amount of visible link previews. -1 means unlimited amount of previews */
+    maxAmountOfPreviews: PropTypes.number,
+};
+
+const defaultProps = {
+    linkMetadata: [],
+    maxAmountOfPreviews: -1,
+};
+
+const LinkPreviewer = (props) => {
```

Comment:
https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#should-i-declare-my-components-with-arrow-functions-const-or-the-function-keyword

We should use function syntax for the functional components